sotodlib/hwp/g3thwp.py

```diff
@@ -1385,6 +1391,20 @@ def _irig_quality_check(self):
             self._irig_time = np.array([])
             self._rising_edge = np.array([])
 
+    def _fix_irig_desync(self):
+        """ Fix IRIG desynchronization by adding constant time offset """
+        if self._irig_desync is None:
+            return
+
+        if len(self._irig_time) < 1:
+            return
```

Comment:
In an "and", Python won't evaluate second condition if first one is False.  So it's safe to:
```
if self._irig_desync is None or len(self._irig_time) == 0:
  return
```

But you really don't need the second check, since the loop over an empty list will do it for you.